### PR TITLE
[codestyle] removed quotes for converting to string

### PR DESCRIPTION
**Pull request by akshitdewan**

### CODE REVIEW

**Comment by vmassol** [comment]
Thanks. Could you link to the sonarqube rule for this? I'd like to check what they say to validate the solution (there are various ways to implement this).

**Comment by akshitdewan** [comment]
Sure! 
http://sonar.xwiki.org/issues/search#issues=0198a3b8-58a9-4dae-95b1-3ec7d6065338
http://sonar.xwiki.org/issues/search#issues=a2d0f74c-b357-4576-b4b7-db82e5948bec
http://sonar.xwiki.org/issues/search#issues=74726c94-0561-4d0f-84fa-369288df5781
http://sonar.xwiki.org/issues/search#issues=df4e0927-b1ad-47c4-8b68-893f9042ab85
http://sonar.xwiki.org/issues/search#issues=d6fca664-9736-4c33-8857-11dd638c6f27

**Comment by akshitdewan** [comment]
they did not mention implementation in these issues, but mentioned in a different one : http://sonar.xwiki.org/issues/search#issues=5faf3edf-357e-45c0-84cd-36411c4948d8

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/reference/InterWikiReferenceTypeSerializer.java [hunk 1]
```diff
@@ -45,27 +45,27 @@
      * Escapes to add when rendering a link reference part.
      */
     private static final String[] ESCAPE_REPLACEMENTS_REFERENCE = new String[] {
-        "" + XWiki20LinkReferenceParser.ESCAPE_CHAR + XWiki20LinkReferenceParser.ESCAPE_CHAR };
+        String.valueOf(XWiki20LinkReferenceParser.ESCAPE_CHAR + XWiki20LinkReferenceParser.ESCAPE_CHAR) };
```
Comment (tmortagne) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/reference/InterWikiReferenceTypeSerializer.java]: Should be `String.valueOf(XWiki20LinkReferenceParser.ESCAPE_CHAR) + XWiki20LinkReferenceParser.ESCAPE_CHAR` I think

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/reference/InterWikiReferenceTypeSerializer.java [hunk 2]
```diff
@@ -45,27 +45,27 @@
...
-        "" + XWiki20LinkReferenceParser.ESCAPE_CHAR };
+        String.valueOf(XWiki20LinkReferenceParser.ESCAPE_CHAR) };
 
     /**
      * Escapes to add when rendering a link query string, anchor or interwiki part.
      */
     private static final String[] ESCAPE_REPLACEMENTS_EXTRA = new String[] {
-        "" + XWiki20LinkReferenceParser.ESCAPE_CHAR + XWiki20LinkReferenceParser.SEPARATOR_INTERWIKI,
-        "" + XWiki20LinkReferenceParser.ESCAPE_CHAR + XWiki20LinkReferenceParser.ESCAPE_CHAR };
+        String.valueOf(XWiki20LinkReferenceParser.ESCAPE_CHAR + XWiki20LinkReferenceParser.SEPARATOR_INTERWIKI),
```
Comment (tmortagne) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/reference/InterWikiReferenceTypeSerializer.java]: Same comment as above.

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/reference/InterWikiReferenceTypeSerializer.java (hunk 1 above)
Comment (akshitdewan) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/reference/InterWikiReferenceTypeSerializer.java]: you're right - careless mistake on my part. will fix, thanks

**Comment by akshitdewan** [comment]
Fixed

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/reference/InterWikiReferenceTypeSerializer.java (hunk 1 above)
Comment (vmassol) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/reference/InterWikiReferenceTypeSerializer.java]: I don't fully agree with Sonarqube here and I find the original approach with the quote simpler to read and less verbose. In addition the compiler will very likely transform it into a performant solution. But I'm ok with the last solution proposed by Thomas, even though I still find the quote solution more readable. I'll let you guys decide what we want to do here but it's likely that I will do it again the way I wrote it initially.

Comment (vmassol) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/reference/InterWikiReferenceTypeSerializer.java]: Some info here: https://stackoverflow.com/questions/15046668/preferred-way-of-declaring-and-converting-primitives-to-strings

**Comment by tmortagne** [comment]
Thanks for your contribution.

**Pull request merged**